src/DocumentFormat.OpenXml/Packaging/RelationshipErrorHandler.cs

```diff
@@ -79,7 +79,13 @@ internal void Handle(Package package)
                     {
                         using var stream = part.GetStream(FileMode.Open, FileAccess.Write);
 
-                        doc.Save(stream);
+                        // Saving directly to the zip stream can cause issues if there are less bytes being written than the original
+                        using var ms = new MemoryStream();
+                        using var writer = new StreamWriter(ms);
+                        doc.Save(writer);
+                        stream.SetLength(ms.Length);
+                        ms.Position = 0;
+                        ms.CopyTo(stream);
```

Comment:
```suggestion
                        stream.SetLength(0);
                        doc.Save(stream);
```

Reply:
I cloned this locally and got it working by just calling `.SetLength(0)`.

Reply:
Oh why didn't I think of that?! Love it.